### PR TITLE
Added Image Resizes support for original_image_url in events API

### LINKED ISSUE #3813: Image saving and resizing for individual APIs
#3738 ensures we have an endpoint where we can upload temporarily the images and send the temp URL link while POST or PATCH to an API for example events, users. The APIs need to now save the image from the temporary url, resize the images to different sizes and store the urls in db.

~To be done after #3738~ (Edit by @niranjan94)

### PR DESCRIPTION
**Pull request by poush**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 

#### Short description of what this resolves:
Added the methods to create different image sizes and save it and update it on database (currently implemented on events API )

Fixes #3813

### CODE REVIEW

File: app/api/helpers/images.py [hunk 1]
```diff
@@ -9,8 +16,92 @@ def get_image_file_name():
...
+    img.paste(im, (0, 0), im)
+
+
+    if maintain_aspect:
+        width_percent = (basewidth / float(img.size[0]))
+        height_size = int((float(img.size[1]) * float(width_percent)))
+
+    img = img.resize((basewidth, height_size), PIL.Image.ANTIALIAS)
+
+    temp_file_relative_path = 'static/media/temp/' + generate_hash(str(image_file)) + get_image_file_name() + '.jpg'
```
Comment (niranjan94) [inline app/api/helpers/images.py]: `static/media/temp` directory may or may not exist. Ensure you create it if needed.

**Comment by SaptakS** [comment]
@poush I am adding a do-not-merge since I am rearranging the folders and links as mentioned in #3824 . It would be better to merge it after it has been tested with it. I will be sending the PR before tomorrow morning. @niranjan94 does that sound okay?

**Comment by niranjan94** [comment]
@SaptakS sounds good. 👍

**Comment by poush** [comment]
@SaptakS Great!

File: app/api/helpers/images.py (hunk 1 above)
Comment (poush) [inline app/api/helpers/images.py]: Done!

**Comment by poush** [comment]
Please review this.

**Comment by poush** [comment]
Anyone ?

**Comment by poush** [comment]
@niranjan94 Can you please review it again

File: app/api/events.py [hunk 2]
```diff
@@ -262,6 +263,9 @@ def after_create_object(self, event, data, view_kwargs):
         user = User.query.filter_by(id=view_kwargs['user_id']).first()
         uer = UsersEventsRoles(user, event, role)
         save_to_db(uer, 'Event Saved')
+        if data.get('original_image_url'):
+            uploaded_images = create_save_image_sizes(data['original_image_url'], 'event', event.id)
```
Comment (niranjan94) [inline app/api/events.py]: Also, please download the original_image_url and resave it and update the new location/link

File: app/api/events.py [hunk 3]
```diff
@@ -391,13 +394,29 @@ def before_get_object(self, view_kwargs):
                 else:
                     view_kwargs['id'] = None
 
+    def before_update_object(self, event, data, view_kwargs):
+        if data.get('original_image_url') and data['original_image_url'] != event.original_image_url:
```
Comment (niranjan94) [inline app/api/events.py]: Also, please download the original_image_url and resave it and update the new location/link

Comment (poush) [inline app/api/events.py]: @niranjan94  Done

File: app/api/events.py (hunk 2 above)
Comment (poush) [inline app/api/events.py]: @niranjan94 Done

**Comment by niranjan94** [comment]
@poush The base branch history was modified. Please update your branch and fix your commits.

**Comment by poush** [comment]
@niranjan94 I think it is good to go. Can you please review it? After that I will implement for another APIs too

File: app/api/helpers/storage.py [hunk 4]
```diff
@@ -151,9 +153,10 @@ def upload_local(uploaded_file, key, **kwargs):
     uploaded_file.save(file_path)
     file_relative_path = '/' + file_relative_path
     if get_settings()['static_domain']:
-        file_relative_path = get_settings()['static_domain'] + \
+        return get_settings()['static_domain'] + \
             file_relative_path.replace('/static', '')
-    return file_relative_path
+    host = urlparse(request.url).hostname
+    return 'http://' + host + file_relative_path
```
Comment (niranjan94) [inline app/api/helpers/storage.py]: Why are you doing this ? (This => I mean forcing `http`)

**Comment by poush** [comment]
@niranjan94 Please review this.

**Comment by SaptakS** [comment]
@poush travis is failing for this PR.

**Comment by poush** [comment]
@niranjan94 @SaptakS Please review, there are other 3 issues depends on this.

**Pull request merged**